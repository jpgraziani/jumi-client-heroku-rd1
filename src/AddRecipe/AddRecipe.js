import React, { Component } from 'react'
import RecipeForm from '../RecipeForm/RecipeForm'
import ApiContext from '../ApiContext'
import config from '../config'

export default class AddRecipe extends Component {
  static defaultProps = {
    history: {
      push: () => { }
    },
  }
  static contextType = ApiContext;

  handleSubmit = e => {
    e.preventDefault()
    const newRecipe = {
      name: e.target['recipe-name'].value,
      directions: e.target['recipe-content'].value,
      created: new Date(),
      ingredients: e.target['ingredients'].value,
      main_protein: e.target['main_protein'].value,
      calories: e.target['calories'].value,
      protein: e.target['protein'].value
    }
    console.log(newRecipe)
    fetch(`${config.API_ENDPOINT}/recipes`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(newRecipe),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
          return res.json()
      })
      .then(recipe => {
        this.context.addRecipe(recipe)
        this.props.history.push(`/recipe/${recipe.id}`)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    
    return (
      <section className='AddNote'>
        <h2>Create a recipe</h2>
        <RecipeForm onSubmit={this.handleSubmit}>
          <div className='field'>
            <label htmlFor='recipe-name-input'>
              Name
            </label>
            <input type='text' id='recipe-name-input' name='recipe-name' />
          </div>
          <div className='field'>
            <label htmlFor='ingredients'>
              ingredients
            </label>
            <textarea id='ingredients' name='ingredients' />
          </div>
          <div className='field'>
            <label htmlFor='recipe-content-input'>
              directions
            </label>
            <textarea id='recipe-content-input' name='recipe-content' />
          </div>
          <div className='field'>
            <label htmlFor='main_protein'>
            main protein
            </label>
            <input type='textr' id='main_proteins' name='main_protein' />
          </div>
          <div className='field'>
            <label htmlFor='calories'>
              calories
            </label>
            <input type='number' id='calories' name='calories' />
          </div>
          <div className='field'>
            <label htmlFor='protein'>
              protein
            </label>
            <input type='number' id='protein' name='protein' />
          </div>
          
          <div className='buttons'>
            <button type='submit'>
              Add recipe
            </button>
          </div>
        </RecipeForm>
      </section>
    )
  }
}
