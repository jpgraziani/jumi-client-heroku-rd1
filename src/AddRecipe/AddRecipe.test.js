import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddRecipe from './AddRecipe'

describe(`AddRecipe component`, () => {
  const newAddRecipe = [
    {
      "id": "1",
      "name": "Test recipe name",
      "created": "2019-01-03T00:00:00.000Z",
      "directions": "Test description",
      "ingredients": "Test ingredients",
      "main_protein": "Test main_protein",
      "calories": "Test calories",
      "protein": "Test protein"
    },
    {
      "id": "2",
      "name": "Test recipe name",
      "created": "2019-01-03T00:00:00.000Z",
      "directions": "Test description",
      "ingredients": "Test ingredients",
      "main_protein": "Test main_protein",
      "calories": "Test calories",
      "protein": "Test protein"
    },
    {
      "id": "3",
      "name": "Test recipe name",
      "created": "2019-01-03T00:00:00.000Z",
      "directions": "Test description",
      "ingredients": "Test ingredients",
      "main_protein": "Test main_protein",
      "calories": "Test calories",
      "protein": "Test protein"
    },
  ]

  it('renders the complete form', () => {
    const context = { recipes: newAddRecipe}
    const wrapper = shallow(<AddRecipe />, context)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})