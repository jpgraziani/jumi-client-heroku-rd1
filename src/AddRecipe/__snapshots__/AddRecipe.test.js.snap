// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe component renders the complete form 1`] = `
<section
  className="AddNote"
>
  <h2>
    Create a recipe
  </h2>
  <RecipeForm
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="recipe-name-input"
      >
        Name
      </label>
      <input
        id="recipe-name-input"
        name="recipe-name"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="ingredients"
      >
        ingredients
      </label>
      <textarea
        id="ingredients"
        name="ingredients"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="recipe-content-input"
      >
        directions
      </label>
      <textarea
        id="recipe-content-input"
        name="recipe-content"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="main_protein"
      >
        main protein
      </label>
      <input
        id="main_proteins"
        name="main_protein"
        type="textr"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="calories"
      >
        calories
      </label>
      <input
        id="calories"
        name="calories"
        type="number"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="protein"
      >
        protein
      </label>
      <input
        id="protein"
        name="protein"
        type="number"
      />
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        Add recipe
      </button>
    </div>
  </RecipeForm>
</section>
`;
